src/plugins/data/public/query/state_sync/triggers/dataset_changed_trigger.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import { QUERY_STATE_TRIGGER_TYPES, QueryStateTrigger } from '../types';
+
+export const DATASET_CHANGED_TRIGGER: QueryStateTrigger = {
```

Comment:
note: this is based on the `Trigger` interface so that we can eventually onboard to that